src/main/minio.js

```diff
@@ -138,6 +138,16 @@ export class Client {
     // and the connection is https we use x-amz-content-sha256=UNSIGNED-PAYLOAD
     // header for signature calculation.
     this.enableSHA256 = !this.anonymous && !params.secure
+
+    this.reqOptions = {}
+  }
+
+  // Sets the supported request options.
+  setRequestOptions(options) {
+    if (!isObject(options)) {
+      throw new TypeError('request options should be of type "object"')
+    }
+    this.reqOptions = _.pick(options, ['agent', 'ca', 'cert', 'ciphers', 'clientCertEngine', 'crl', 'dhparam', 'ecdhCurve', 'honorCipherOrder', 'key', 'passphrase', 'pfx', 'rejectUnauthorized', 'secureOptions', 'secureProtocol', 'servername', 'sessionIdContext'])
```

Comment:
I am not sure why this is needed, in addition this list needs to be updated when nodejs tls API gets updated.

Reply:
This is needed for developers to pass custom agent or TLS related configs (custom certs) Have a look at the discussion on https://github.com/minio/minio-js/pull/677

Reply:
> This is needed for developers to pass custom agent or TLS related configs (custom certs) Have a look at the discussion on #677

I mean why this is not only `this.reqOptions = options`, why are we picking specific fields ?

Reply:
because we dont want to let the user provide `option.host` `option.headers` etc even by mistake which cause minio-js calls to fail